### DISCUSSION TITLE
Is there any plans on officially releasing 0.6.2 any time soon?

### DISCUSSION THREAD

**Discussion by rexim** [c3lang/c3c#1383] [General]
There is a milestone for it and it seems completed: https://github.com/c3lang/c3c/milestone/9

**Comment by lerno** [comment]
I mostly want to be sure that all known bugs are addressed. There's also this task: #1352 which has caused me a lot of pondering on how I wanted to solve it. Originally it seemed straightforward, but has led into a lot of thoughts on call parameter evaluation ordering and named parameters. I discovered two bugs in the current code:

1. Resolving `$vasplat` slicing may have side effects that does not honor argument left-to-right evaluation ordering
2. Named parameters will not always honor parameter left-to-right ordering.

And this is not taking into consideration what splat would do. So the question is:
1. Take a comprehensive look at named parameters and argument ordering now
2. Just patch the issues (even though it's likely to be something that has to be fixed later)
3. Release with those known issues
4. Try to finish splat without changing named parameters/argument ordering which seems unnecessarily complex for someone wanting to reimplement the compiler.

Also I fixed a bug as recently as today, I'd like to know that there hasn't been a bug spotted in a while before I release, so if I release with known issues, then it'll be in about a week I think.

**Reply by rexim** [reply]
If you knew that it's not gonna be straightforward would you start working on it towards 0.6.2 in the first place? I personally usually step back in such situations and reconsider. But, I mean, you have much more information and context, I'm just sharing my personal experience! All the luck with resolving the problems.

Thank you for the response! I'm just looking forward to the release of the arguments passing to the compiler subcommands, because it's a very crucial QoL feature that prevents me from using the compiler as I need to. Please don't feel pressured take your time and do as you see fit. I'll be patiently waiting.

**Reply by lerno** [reply]
Initially I had forgotten about the problem with ordering, and was happily almost wrapping up the feature when I realized that it needed an overhaul in how the compiler was working with named arguments as and splats. So I tried to tackle this twice, and it just felt bad both times. Which has led me to take a few steps back and even reconsider how named arguments work.

Because quite unrelated I found that I am always annoyed when I have to use a named parameter: `new_create_foo(a, b, .allocator = test_allocator)` is much more annoying to read than `new_create_foo(a, b, test_allocator);`

And a change I was considering quite unrelated was just to have the convention of two different function in case with option allocator arguments, which is:

`new_create_foo(a, b)` - heap allocation
`create_foo(a, b, test_allocator)` - allocate with any allocator ("new_create_foo" calls this one with the heap allocator)

This would remove 98% of all the named argument uses in the stdlib(!)

There's a second function yes, but it just feels more like C.

So maybe named arguments should go? I don't feel they're getting used for good stuff. Optional arguments with default values are fine. But named arguments are kind of "well, we want to make optional arguments even more convenient because it's used often" – something which might be true for languages with *constructors* where you're often encouraged to put every single property in the constructor. But that's not really how C does it.

But enough of rambling, I'm thinking about it seriously but no decision has been made. Any removal of named arguments would need to be in a 0.7.0 anyway, not 0.6.x.

So from just implementing splat to now thinking of removing named arguments. So if you want C3 to keep them, now's a good time to speak out in their defense :D
